### DISCUSSION TITLE
how to retrieve workspace root from .vimrc

### DISCUSSION THREAD

**Discussion by rameshsanth** [tpope/vim-fugitive#1805] [Q&A] [answered]
This is a basic VIM question.. I'm stumped and cannot figure this out..

I would like to retrieve the Git workspace root and setup the "path" variable appropriately from .vimrc
i tried this.. but keep getting errors

let s = call('FugitiveWorkTree', [])
set path+= s . '/src', s . '/test'

But i get this error.
line  989:
E117: Unknown function: FugitiveWorkTree

Its clear i'm unable to access the Fugitve functions from .vimrc.. 
I even looked into plugins like lightline how they manage to call Fugitive functions.. just cant figure it out..

Lightline does something like this.. 
  for f in values(s:lightline.component_function)
       silent! call call(f, [])
  endfor

**Comment by tpope** [comment, marked as answer]
The reason you can't call `FugitiveWorkTree()` is because Vim loads your vimrc before it loads plugins. Rather than attempt to work around this, I am instead inclined to direct you to [projectionist.vim](/tpope/vim-projectionist), which provides a robust implementation of the exact behavior you're looking for. Here's an example config that does what you want:

```
let g:projectionist_heuristics = {
      \ '.git/': {'path': ['src', 'test']}
      \ }
```

Even better, change `.git/` to something more specific, like `pom.xml`.
